File: Clg_Placement/Java/_19.ChechIfTwoArraysAreEqualOrNot.java
class Solution{
    //Function to check if two arrays are equal or not.
    public static boolean check(long A[],long B[],int N)
    {
        HashMap<Long,Integer> mpp = new HashMap<>();
        for(long num: A){
            mpp.put(num,mpp.getOrDefault(num,0)+1);
        }
        for(long it: B){
            if(!mpp.containsKey(it) || mpp.get(it)==0) return false;
            mpp.put(it,mpp.get(it)-1);
        }
        return true;
    }
}

File: Clg_Placement/C++/Arrays/_19.ChechIfTwoArraysAreEqualOrNot.cpp
//better:  time:O(NlogN)  coz of sorting   Space: O(1)
class Solution{
    public:

    //Function to check if two arrays are equal or not.
    bool check(vector<ll> A, vector<ll> B, int N) {
        sort(A.begin(),A.end());
        sort(B.begin(),B.end());
        for(int i=0;i<N;i++){
            if(A[i]!=B[i]) return 0;
        }
        return 1;
    }
};

//optimal: time:O(N) space:O(N)   coz of unordered_map
class Solution{
    public:

    //Function to check if two arrays are equal or not.
    bool check(vector<ll> A, vector<ll> B, int N) {
        unordered_map<ll,int> mpp;   //ll means long long c++ suports this short form.
        for(ll num:A){
            mpp[num]++;
        }
        for(ll num:B){
            if(mpp.find(num)==mpp.end()||mpp[num]==0){
                return false;
            }
            mpp[num]--;
        }
        return true;
    }
};
